memorious/operations/aleph.py

```diff
@@ -8,69 +9,75 @@
 from alephclient.util import backoff
 from alephclient.errors import AlephException
 from memorious.core import get_rate_limit  # type: ignore
+from memorious.logic.context import Context
+from memorious.logic.meta import Meta
 
 
-def _create_document_metadata(context, data) -> dict:
-    meta = {}
-    languages = context.params.get("languages")
-    meta["languages"] = data.get("languages", languages)
-    countries = context.params.get("countries")
-    meta["countries"] = data.get("countries", countries)
-    mime_type = context.params.get("mime_type")
-    meta["mime_type"] = data.get("mime_type", mime_type)
-    return meta
+def _create_document_metadata(meta_base: Meta, context: Context, data: dict) -> Meta:
+    languages: list[str] = list(context.params.get("languages", []))
+    countries: list[str] = list(context.params.get("countries", []))
+    mime_type: str = context.params.get("mime_type", "")
+
+    context.log.warn(languages)
+    meta_base["languages"] = data.get("languages", languages)
+    meta_base["countries"] = data.get("countries", countries)
+    meta_base["mime_type"] = data.get("mime_type", mime_type)
+
+    return meta_base
 
 
-def _create_meta_object(context, data) -> dict:
+def _create_meta_object(context: Context, data: dict) -> Meta:
     source_url = data.get("source_url", data.get("url"))
     foreign_id = data.get("foreign_id", data.get("request_id", source_url))
 
-    meta = {
-        "crawler": context.crawler.name,
-        "foreign_id": foreign_id,
-        "source_url": source_url,
-        "title": data.get("title"),
-        "author": data.get("author"),
-        "publisher": data.get("publisher"),
-        "file_name": data.get("file_name"),
-        "retrieved_at": data.get("retrieved_at"),
-        "modified_at": data.get("modified_at"),
-        "published_at": data.get("published_at"),
-        "headers": data.get("headers", {}),
-        "keywords": data.get("keywords", []),
-    }
+    meta = Meta(
+        crawler=context.crawler.name,
+        foreign_id=foreign_id,
+        source_url=source_url,
+        title=data.get("title"),
+        author=data.get("author"),
+        publisher=data.get("publisher"),
+        file_name=data.get("file_name"),
+        retrieved_at=data.get("retrieved_at"),
+        modified_at=data.get("modified_at"),
+        published_at=data.get("published_at"),
+        headers=data.get("headers", {}),
+        keywords=data.get("keywords", []),
+    )
 
     if data.get("aleph_folder_id"):
         meta["parent"] = {"id": data.get("aleph_folder_id")}
```

Comment:
We are fixing up 2 styles here. We are treating meta as a object in the line above and as a dict here.

And imo we can merge `_create_meta_object` and `_create_document_metadata` together into just one function and only set whatever metadata is available.